File: app/src/main/java/com/cjmobileapps/quidditchplayersandroid/data/model/House.kt
package com.cjmobileapps.quidditchplayersandroid.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class House(
    @PrimaryKey(autoGenerate = true) val houseId: Int = 0,
    val name: HouseName,
    val imageUrl: String,
    val emoji: String,
)

enum class HouseName {
    GRYFFINDOR,
    SLYTHERIN,
    RAVENCLAW,
    HUFFLEPUFF,
}

File: app/src/main/cpp/src/com/cjmobileapps/quidditchplayers/model/House.cpp
#include "House.h"

namespace com::cjmobileapps::quidditchplayers::model {
    // Constructor
    House::House(int houseId, HouseName name, const std::string& imageUrl, const std::string& emoji)
        : houseId(houseId), name(name), imageUrl(imageUrl), emoji(emoji) {}

    // Getters
    int House::getHouseId() const {
        return houseId;
    }

    HouseName House::getName() const {
        return name;
    }

    std::string House::getImageUrl() const {
        return imageUrl;
    }

    std::string House::getEmoji() const {
        return emoji;
    }

    // Setters
    void House::setHouseId(int houseId) {
        this->houseId = houseId;
    }

    void House::setName(HouseName name) {
        this->name = name;
    }

    void House::setImageUrl(const std::string& imageUrl) {
        this->imageUrl = imageUrl;
    }

    void House::setEmoji(const std::string& emoji) {
        this->emoji = emoji;
    }
}
